nsxt/resource_nsxt_policy_tier0_gateway_gre_tunnel.go

```diff
@@ -0,0 +1,409 @@
+				PrefixLen:   &prefixLen,
+			}
+			tunnelInterfaceSubnets = append(tunnelInterfaceSubnets, elem)
+		}
+		elem := model.TunnelAddress{
+			EdgePath:              &edgePath,
+			SourceAddress:         &sourceAddress,
+			TunnelInterfaceSubnet: tunnelInterfaceSubnets,
+		}
+		tunnelAddresses = append(tunnelAddresses, elem)
+	}
+	var tunnelKeepalive model.TunnelKeepAlive
+	tunnelKeepAliveList := d.Get("tunnel_keepalive").([]interface{})
+	for _, tka := range tunnelKeepAliveList {
+		tkaElement := tka.(map[string]interface{})
+		deadTimeMultiplier := int64(tkaElement["dead_time_multiplier"].(int))
+		enableKeepaliveAck := tkaElement["enable_keepalive_ack"].(bool)
+		tkaEnabled := tkaElement["enabled"].(bool)
+		keepaliveInterval := int64(tkaElement["keepalive_interval"].(int))
+
+		tunnelKeepalive = model.TunnelKeepAlive{
+			DeadTimeMultiplier: &deadTimeMultiplier,
+			EnableKeepaliveAck: &enableKeepaliveAck,
+			Enabled:            &tkaEnabled,
+			KeepaliveInterval:  &keepaliveInterval,
+		}
+		break // Only one element is allowed.
+	}
+
+	greTunnel := model.GreTunnel{
+		DisplayName:        &displayName,
+		Description:        &description,
+		Tags:               tags,
+		Enabled:            &enabled,
+		Mtu:                &mtu,
+		DestinationAddress: &destinationAddress,
+		TunnelAddress:      tunnelAddresses,
+		TunnelKeepalive:    &tunnelKeepalive,
+		ResourceType:       model.GreTunnel__TYPE_IDENTIFIER,
+	}
+	dataValue, errs := converter.ConvertToVapi(greTunnel, model.GreTunnelBindingType())
+	if errs != nil {
+		return nil, errs[0]
+	} else if dataValue != nil {
+		return dataValue.(*data.StructValue), nil
+	}
+	return nil, nil
+
+}
+
+func resourceNsxtPolicyTier0GatewayGRETunnelExists(id, localeServicePath string, connector client.Connector, isGlobalManager bool) (bool, error) {
+	isT0, tier0id, localeSvcID, err := parseLocaleServicePolicyPath(localeServicePath)
+	if err != nil {
+		return false, err
+	}
+	if !isT0 {
+		return false, fmt.Errorf("locale service path %s is not associated with a tier0 router", localeServicePath)
+	}
+
+	client := locale_services.NewTunnelsClient(connector)
+	_, err = client.Get(tier0id, localeSvcID, id)
+
+	if isNotFoundError(err) {
+		return false, nil
+	}
+	return true, nil
+}
+
+func resourceNsxtPolicyTier0GatewayGRETunnelCreate(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	id := d.Get("nsx_id").(string)
+	if id == "" {
+		id = newUUID()
+	}
+	localeServicePath := d.Get("locale_service_path").(string)
+	isT0, tier0id, localeSvcID, err := parseLocaleServicePolicyPath(localeServicePath)
+	if err != nil {
+		return handleCreateError("GreTunnel", id, err)
+	}
+	if !isT0 {
+		return fmt.Errorf("locale service path %s is not associated with a tier0 router", localeServicePath)
+	}
+	client := locale_services.NewTunnelsClient(connector)
+
+	obj, err := tier0GatewayGRETunnelFromSchema(d)
+	if err != nil {
+		return fmt.Errorf("failed to create GRE Tunnel: %v", err)
```

Comment:
nit abou the error message: we have not yet tried the creation here, so maybe the error should state "unable to create the GRE tunnel" or something like that